File: lib_video/src/main/java/com/effective/android/video/bean/VideoCache.java
package com.effective.android.video.bean;

import androidx.annotation.NonNull;

/**
 * 定义单例加载的视频信息及缓存信息
 * created by yummylau 2019/04/20
 */
public class VideoCache {

    public long id;
    public boolean isCache;
    public boolean isLoop;
    public boolean isMute;
    public boolean isContinuePlaySame;

    public long lastPosition;
    public VideoStatus lastStatus;

    public VideoInfo videoInfo;

    public VideoCache(Builder builder) {
        this.isCache = builder.isCache;
        this.isLoop = builder.isLoop;
        this.isMute = builder.isMute;
        this.isContinuePlaySame = builder.isContinuePlaySame;
        this.videoInfo = builder.videoInfo;
        this.id = System.currentTimeMillis();
    }


    public static class Builder {
        private boolean isCache = true;               //是否缓存
        private boolean isMute = true;                 //是否静音
        private boolean isLoop = true;                 //是否循环播放
        private boolean isContinuePlaySame = true;     //是否继续播放，两个页面相同视频
        private VideoInfo videoInfo;

        public Builder cache(boolean cache) {
            this.isCache = cache;
            return this;
        }

        public Builder loop(boolean loop) {
            this.isLoop = loop;
            return this;
        }

        public Builder mute(boolean mute) {
            this.isMute = mute;
            return this;
        }

        public Builder continuePlaySame(boolean continuePlaySame) {
            this.isContinuePlaySame = continuePlaySame;
            return this;
        }

        public VideoCache build(@NonNull VideoInfo videoInfo) {
            this.videoInfo = videoInfo;
            return new VideoCache(this);
        }
    }

}

File: lib_video/src/main/java/com/effective/android/video/bean/VideoCache.kt
package com.effective.android.video.bean

import android.text.TextUtils
import com.effective.android.video.core.VideoReceivingLayer

/**
 * 定义单例加载的视频信息及缓存信息
 * created by yummylau 2019/04/20
 */
class VideoCache(builder: Builder) {

    var id: Long = 0
    var isCache: Boolean = false
    var isLoop: Boolean = false
    var isMute: Boolean = false
    var isContinuePlaySame: Boolean = false

    var lastPosition: Long = 0
    var lastStatus: VideoStatus? = null

    val videoInfo: VideoInfo
    val receivingLayer: VideoReceivingLayer


    init {
        this.isCache = builder.isCache
        this.isLoop = builder.isLoop
        this.isMute = builder.isMute
        this.isContinuePlaySame = builder.isContinuePlaySame
        this.videoInfo = builder.videoInfo!!
        this.receivingLayer = builder.receivingLayer!!
        this.id = System.currentTimeMillis()
    }

    fun isSameContent(videoCache: VideoCache?): Boolean {
        return videoCache != null && TextUtils.equals(videoCache.videoInfo!!.url, videoInfo!!.url)
    }

    class Builder {
        var isCache = true               //是否缓存
        var isMute = true                 //是否静音
        var isLoop = true                 //是否循环播放
        var isContinuePlaySame = true     //是否继续播放，两个页面相同视频
        var videoInfo: VideoInfo? = null
        var receivingLayer: VideoReceivingLayer? = null

        fun cache(cache: Boolean): Builder {
            this.isCache = cache
            return this
        }

        fun loop(loop: Boolean): Builder {
            this.isLoop = loop
            return this
        }

        fun mute(mute: Boolean): Builder {
            this.isMute = mute
            return this
        }

        fun continuePlaySame(continuePlaySame: Boolean): Builder {
            this.isContinuePlaySame = continuePlaySame
            return this
        }

        fun build(videoInfo: VideoInfo, receivingLayer: VideoReceivingLayer): VideoCache {
            this.videoInfo = videoInfo
            this.receivingLayer = receivingLayer
            return VideoCache(this)
        }
    }

}
